cxx/distributions/dirichlet_categorical.hh

```diff
@@ -0,0 +1,62 @@
+// Copyright 2024
+// See LICENSE.txt
+
+#pragma once
+#include "base.hh"
+
+class DirichletCategorical : public Distribution {
+public:
+    double alpha = 1;  // hyperparameter (applies to all categories)
+    std::vector<int> counts;  // counts of observed categories
+    PRNG *prng;
+
+    DirichletCategorical(PRNG *prng, int k) {  // k is number of categories
+        this->prng = prng;
+        counts = std::vector<int>(k, 0);
+    }
+    void incorporate(double x) {
```

Comment:
At first I wondered why this was `double`, but I see that the `Distribution` base class assumes that values are always of `double` type. I think we will very soon have to generalize `Distribution` to support other value types (e.g. strings), and at that point, we may want to come back and make this an integer-valued distribution. But for now this looks good!

Reply:
Added an issue for this (#3 )